File: modules/01-basic-syntax/python/main.py
import src.utils.utils as utils
import src.modules.compound_types as compound_types
import src.modules.references as references
import src.modules.slices as slices
import src.modules.functions as functions

def main():
  utils.print_h1(title="Basic syntax")

  # Compound types
  utils.print_h2(title="Compound types")
  compound_types.array_asignment_and_access()
  compound_types.tuple_assignment_and_access()
  utils.print_br()

  # References
  utils.print_h2(title="References")
  references.references()
  utils.print_br()

  # Slices
  utils.print_h2(title="Slices")
  slices.array_slices()
  slices.string_vs_str()
  utils.print_br()

  # Functions
  utils.print_h2(title="Functions")
  functions.fizzbuzz_to(10)
  functions.test_methods_from_rectangle()
  functions.test_overloading()
  utils.print_br()

if __name__ == "__main__":
  main()

File: modules/01-basic-syntax/typescript/main.ts
import * as utils from './src/utils/utils';
import * as compound_types from './src/modules/compound_types';
import * as references from './src/modules/references';
import * as slices from './src/modules/slices';
import * as functions from './src/modules/functions';

utils.print_h1('Basic syntax');

// Compound types
utils.print_h2('Compound types');
compound_types.array_asignment_and_access();
compound_types.tuple_assignment_and_access();
utils.print_br();

// References
utils.print_h2('References');
references.references();
utils.print_br();

// Slices
utils.print_h2('Slices');
slices.array_slices();
slices.string_vs_str();
utils.print_br();

// Functions
utils.print_h2('Functions');
functions.fizzbuzz_to(10);
functions.test_methods_from_rectangle();
functions.test_overloading();
utils.print_br();
